File: old_trunk/cost/iona/orbacus/cpp/main/TestIntfLongLong_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestIntfLongLong_impl.h>

#include <limits.h>

using namespace CORBA;

TestIntfLongLong_impl::TestIntfLongLong_impl()
{
}

LongLong
TestIntfLongLong_impl::attrLongLong()
    throw (CORBA::SystemException)
{
    return m_aLongLong;
}

void
TestIntfLongLong_impl::attrLongLong(
    LongLong val
)
    throw (CORBA::SystemException)
{
    m_aLongLong = val;
}

LongLong
TestIntfLongLong_impl::opLongLong(
    LongLong a0,
    LongLong& a1,
    LongLong& a2
)
    throw (CORBA::SystemException)
{
    m_aLongLong = a0 + a1;
    a1 = a2 = m_aLongLong;
    return m_aLongLong;
}

LongLong
TestIntfLongLong_impl::opLongLongEx(
    LongLong a0,
    LongLong& a1,
    LongLong&
)
    throw (ORBTest_LongLong::ExLongLong,
	   CORBA::SystemException)
{
    m_aLongLong = a0 + a1;
    throw ORBTest_LongLong::ExLongLong(m_aLongLong);
    return 0; // Some compilers need this
}

ULongLong
TestIntfLongLong_impl::attrULongLong()
    throw (CORBA::SystemException)
{
    return m_aULongLong;
}

void
TestIntfLongLong_impl::attrULongLong(
    ULongLong val
)
    throw (CORBA::SystemException)
{
    m_aULongLong = val;
}

ULongLong
TestIntfLongLong_impl::opULongLong(
    ULongLong a0,
    ULongLong& a1,
    ULongLong& a2
)
    throw (CORBA::SystemException)
{
    m_aULongLong = a0 + a1;
    a1 = a2 = m_aULongLong;
    return m_aULongLong;
}

ULongLong
TestIntfLongLong_impl::opULongLongEx(
    ULongLong a0,
    ULongLong& a1,
    ULongLong&
)
    throw (ORBTest_LongLong::ExULongLong,
	   CORBA::SystemException)
{
    m_aULongLong = a0 + a1;
    throw ORBTest_LongLong::ExULongLong(m_aULongLong);
    return 0; // Some compilers need this
}

File: cost/legacy/iona/orbacus/java/main/TestIntfLongLong_impl.java
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package ORBTest;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;

final class TestIntfLongLong_impl
    extends ORBTest_LongLong.IntfPOA
{
    private POA m_poa;
    private long m_aLongLong;
    private long m_aULongLong;

    public
    TestIntfLongLong_impl(
	POA poa
    )
    {
	m_poa = poa;
    }

    public synchronized long
    attrLongLong()
    {
	return m_aLongLong;
    }

    public synchronized void
    attrLongLong(
	long value
    )
    {
	m_aLongLong = value;
    }

    public synchronized long
    opLongLong(
	long a0,
	LongHolder a1,
	LongHolder a2
    )
    {
	m_aLongLong = a0 + a1.value;
	a1.value = a2.value = m_aLongLong;
	return m_aLongLong;
    }

    public synchronized long
    opLongLongEx(
	long a0,
	LongHolder a1,
	LongHolder a2
    )
        throws ORBTest_LongLong.ExLongLong
    {
	m_aLongLong = a0 + a1.value;
	throw new ORBTest_LongLong.ExLongLong(m_aLongLong);
    }

    public synchronized long
    attrULongLong()
    {
	return m_aULongLong;
    }

    public synchronized void
    attrULongLong(
	long value
    )
    {
	m_aULongLong = value;
    }

    public synchronized long
    opULongLong(
	long a0,
	LongHolder a1,
	LongHolder a2
    )
    {
	m_aULongLong = a0 + a1.value;
	a1.value = a2.value = m_aULongLong;
	return m_aULongLong;
    }

    public synchronized long
    opULongLongEx(
	long a0,
	LongHolder a1,
	LongHolder a2
    )
        throws ORBTest_LongLong.ExULongLong
    {
	m_aULongLong = a0 + a1.value;
	throw new ORBTest_LongLong.ExULongLong(m_aULongLong);
    }

    public org.omg.PortableServer.POA
    _default_POA()
    {
	return m_poa;
    }
}
